src/cryptonote_core/blockchain.cpp

```
@@ -3322,18 +3517,68 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
     crypto::hash tx_prefix_hash = get_transaction_prefix_hash(tx);
 
     std::vector<std::vector<rct::ctkey>> pubkeys(tx.vin.size());
+    // HF21: parallel ring of blinded asset ids, only populated for zarcanum inputs.
+    std::vector<std::vector<crypto::asset_id>> zc_asset_id_rings(tx.vin.size());
+    std::vector<size_t> native_sig_indices; // which sig_index values are txin_to_key, in order
+    native_sig_indices.reserve(tx.vin.size());
     size_t sig_index = 0;
     const crypto::key_image *last_key_image = NULL;
     for (size_t sig_index = 0; sig_index < tx.vin.size(); sig_index++)
```

Clang (14.0.6)

[note] +2, including nesting penalty of 1, nesting level increased to 2

(clang)